ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/newLayout.tsx

```diff
@@ -0,0 +1,144 @@
+'use client';
+
+import { useCallback, useState } from 'react';
+import { Badge } from '@inngest/components/Badge';
+import { InvokeModal } from '@inngest/components/InvokeButton';
+import { Skeleton } from '@inngest/components/Skeleton/Skeleton';
+import { RiPauseCircleLine } from '@remixicon/react';
+import { useMutation } from 'urql';
+
+import { ArchivedAppBanner } from '@/components/ArchivedAppBanner';
+import { useEnvironment } from '@/components/Environments/environment-context';
+import { ActionsMenu } from '@/components/Functions/ActionMenu';
+import { CancelFunctionModal } from '@/components/Functions/CancelFunction/CancelFunctionModal';
+import { PauseFunctionModal } from '@/components/Functions/PauseFunction/PauseModal';
+import { Header } from '@/components/Header/Header';
+import { graphql } from '@/gql';
+import { useFunction } from '@/queries';
+import NewReplayModal from './logs/NewReplayModal';
+
+const InvokeFunctionDocument = graphql(`
+  mutation InvokeFunction($envID: UUID!, $data: Map, $functionSlug: String!) {
+    invokeFunction(envID: $envID, data: $data, functionSlug: $functionSlug)
+  }
+`);
+
+type FunctionLayoutProps = {
+  children: React.ReactNode;
+  params: {
+    environmentSlug: string;
+    slug: string;
+  };
+};
+
+export default function FunctionLayout({
+  children,
+  params: { environmentSlug, slug },
+}: FunctionLayoutProps) {
+  const [invokOpen, setInvokeOpen] = useState(false);
+  const [pauseOpen, setPauseOpen] = useState(false);
+  const [cancelOpen, setCancelOpen] = useState(false);
+  const [replayOpen, setReplayOpen] = useState(false);
+
+  const functionSlug = decodeURIComponent(slug);
+  const [{ data, fetching }] = useFunction({ functionSlug });
+  const [, invokeFunction] = useMutation(InvokeFunctionDocument);
+  const env = useEnvironment();
+
+  const fn = data?.workspace.workflow;
+  const { isArchived = false, isPaused } = fn ?? {};
+
+  const doesFunctionAcceptPayload =
+    fn?.current?.triggers.some((trigger) => {
+      return trigger.eventName;
+    }) ?? false;
+
+  const invokeAction = useCallback(
+    (data: Record<string, unknown>) => {
+      invokeFunction({
+        envID: env.id,
+        data,
+        functionSlug,
+      });
+      setInvokeOpen(false);
+    },
+    [env.id, functionSlug, invokeFunction]
+  );
+
+  const externalAppID = data?.workspace.workflow?.appName;
+
+  return (
+    <>
+      {externalAppID && <ArchivedAppBanner externalAppID={externalAppID} />}
+      {invokOpen && (
+        <InvokeModal
```

Comment:
Extracted all the action modals from the buttons so we can invoke them from a menu.